prboom2/src/r_defs.h

```diff
@@ -390,23 +390,20 @@ typedef struct msecnode_s
 typedef struct
 {
   vertex_t *v1, *v2;
-  fixed_t offset;
```

Comment:
There is at least one struct in the doom code base that has members that need to be in specific positions (as insane as that sounds). I don't think this is one, but I wanted to bring that up as a caution for you if you hadn't considered it.

Reply:
Do you have any idea which one and for what reason?

Reply:
```c
typedef struct visplane
{
  struct visplane *next;        // Next visplane in hash chain -- killough
  int picnum, lightlevel, minx, maxx;
  int special; // heretic
  fixed_t height;
  fixed_t xoffs, yoffs;         // killough 2/28/98: Support scrolling flats
  angle_t rotation;
  fixed_t xscale;
  fixed_t yscale;
  // e6y: resolution limitation is removed
  // bottom and top arrays are dynamically
  // allocated immediately after the visplane
  unsigned short *bottom;
  unsigned short pad1;          // leave pads for [minx-1]/[maxx+1]
  unsigned short top[3];

  // NEW FIELDS MUST BE ADDED **ABOVE** `unsigned short *bottom;`

} visplane_t;
```
The comment at the bottom is from me, after I ran into rendering errors while adding udmf transforms to this struct. The visplane has a dynamic size based on resolution, which shows up here:
```c
  visplane_t *check = freetail;
  if (!check)
  {
    // e6y: resolution limitation is removed
    check = Z_Calloc(1, sizeof(*check) + sizeof(*check->top) * (SCREENWIDTH * 2));
```

Reply:
Ah, yeah.  This would typically be done with "flexible array members" in C99 or later, e.g.

```c
struct foo
{
    ...
    unsigned short array[];
}
```

If you try to put something after `array` the compiler will complain.